File: src/main/kotlin/dev/jefy/connectpro/management/domain/repositoty/AwardRepository.java
package dev.jefy.connectpro.management.domain.repositoty;

import org.springframework.stereotype.Repository;

import dev.jefy.connectpro.management.domain.Award;
import dev.jefy.connectpro.management.domain.vo.AwardId;
import dev.jefy.connectpro.shared.infrastructure.ddd.AggregateRepository;

/**
 * @author Jôph Yamba
 */
@Repository
public interface AwardRepository extends AggregateRepository<Award, AwardId> {
    boolean existsByName(String name);
}

File: src/main/kotlin/dev/jefy/connectpro/management/domain/repository/AwardRepository.kt
package dev.jefy.connectpro.management.domain.repository

import org.springframework.stereotype.Repository
import dev.jefy.connectpro.management.domain.Award
import dev.jefy.connectpro.management.domain.vo.AwardId
import org.springframework.data.jpa.repository.JpaRepository

@Repository
interface AwardRepository : JpaRepository<Award, AwardId> {
    fun existsByName(name: String): Boolean
}